tests/kohana/Log/FileTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/**
+ * Tests Kohana Logging API - writing to file
+ *
+ * @group kohana
+ * @group kohana.core
+ * @group kohana.core.logging
+ * @group kohana.core.logging.file
+ *
+ * @package    Kohana
+ * @category   Tests
+ * @author     Kohana Team
+ * @copyright  (c) 2008-2012 Kohana Team
+ * @license    http://kohanaframework.org/license
+ */
+class Kohana_Log_FileTest extends Unittest_TestCase {
+
+	private $log_dir;
+
+	/**
+	 * Sets up the test enviroment
+	 */
+	// @codingStandardsIgnoreStart
+	function setUp()
+	// @codingStandardsIgnoreEnd
+	{
+		parent::setUp();
+
+		$this->log_dir = sys_get_temp_dir()
+		  . DIRECTORY_SEPARATOR
+		  . 'kohana_test_logs-' . microtime(TRUE);
+
+		mkdir($this->log_dir, 02777);
+
+		// Reset permissions to fix umask issues
+		chmod($this->log_dir, 02777);
+	}
+
+	/**
+	 * Provider for test_file
+	 *
+	 */
+	public function provider_file()
+	{
+		// a sample exception used in data sets
+		$exception = new Exception('dummy exception text');
+
+		return [
+			// data set #0
+			[
+				// raw log messages
+				[
+					[
+						'time' => strtotime('2015-11-19 10:05:48'),
+						'level' => \Psr\Log\LogLevel::DEBUG,
+						'body' => 'dummy text',
+						'line' => 10,
+						'file' => '/path/to/file.php',
+					]
+				],
+				// expected file content
+<<<'LOG'
+<?php exit; ?>
+
+2015-11-19 10:05:48 --- DEBUG: dummy text in /path/to/file.php:10
+
+LOG
+
+			],
+			// data set #1
+			[
+				// raw log messages
+				[
+					[
+						'time' => strtotime('2015-11-19 10:05:48'),
+						'level' => \Psr\Log\LogLevel::ERROR,
+						'body' => 'another dummy text',
+						'file' => '/path/to/another/file.php',
+						'line' => 20,
+						'exception' => $exception,
+					]
+				],
+				// expected file content
+<<<'LOG'
+<?php exit; ?>
+
+2015-11-19 10:05:48 --- ERROR: another dummy text in /path/to/another/file.php:20
+2015-11-19 10:05:48 --- DEBUG: 
+LOG
+				. $exception->getTraceAsString()
+				. ' in /path/to/another/file.php:20'
+				. PHP_EOL
+			]
+		];
+	}
+
+	/**
+	 * Tests logging with the file writer
+	 *
+	 * @test
+	 * @dataProvider provider_file
+	 */
+	public function test_file(array $log_entries, $expected)
```

Comment:
Tested that a single writer instance appends to an existing file https://github.com/kohana/core/commit/28087908932f380b813dd9393f56ef1e44d6accc